<table align="left">
  <td>
    <a href="https://colab.research.google.com/github/marco-canas/fundamentos_de_programacion/blob/main/2_clases/unidad1/2_chapter.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>
  </td>
  <td>
    <a target="_blank" href="https://kaggle.com/kernels/welcome?src=https://github.com/marco-canas/fundamentos_de_programacion/blob/main/2_clases/unidad1/2_chapter.ipynb"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" /></a>
  </td>
</table>

### [Video de apoyo a la lectura interactiva y experimental de este cuaderno](https://studio.youtube.com/video/GpJTyzxyPyE/edit)

### [Vínculo al programa del curso de Fundamentos de programación:](https://github.com/marco-canas/fundamentos_de_programacion/blob/main/programa/1_Programa_Fundamentos_Programacion.md)



## Código del curso: 314029 

Un reconocimiento a mis estudiantes que han construido conmigo este saber pedagógico:

<img src = 'https://github.com/marco-canas/algebra_y_trigonometria/blob/main/7_images/5_clase_3_2025-08-12_hora_11_04.jpg?raw=true'> 



In [4]:
import pandas as pd
import numpy as np

# Cargar la lista de estudiantes desde el archivo CSV
path = 'C:/Users/marco/Documentos/docencia/groups_list/fundamentos_programacion_314029_35.xlsx'
df = pd.read_excel(path)
df.index = np.arange(1, len(df) + 1)  # Cambiar el índice para que empiece en 1
df.head(17)


,Cédula,Nombre,Email
1,1038111190,CANOLE PEÑATE SEBASTIAN,sebastian.canolep@udea.edu.co
2,1048073285,CAPACHERO AGAMEZ JUAN PABLO,juan.capachero@udea.edu.co
3,1032251814,CASTRO ERAZO VALERIA,v.castro1@udea.edu.co
4,1038111332,CAUSIL ACOSTA ANDREW DAVID,a.causil@udea.edu.co
5,1063277880,CLEMENTE APARICIO JULIANA,juliana.clemente@udea.edu.co
6,1066604944,CLEMENTE APARICIO SAMUEL,s.clemente@udea.edu.co
7,1066600241,DURANGO TIRADO DANIEL DARIO,daniel.durango1@udea.edu.co
8,1038648760,MARIN VILLARREAL PABLO ALEJANDRO,pablo.marin1@udea.edu.co
9,1040515084,MOSQUERA TOSCANO STEFHANY PATRICIA,stefhany.mosquera@udea.edu.co
10,1126603686,ORTEGA QUINTERO GUILLERMO ANDRES,guillermo.ortegaq@udea.edu.co


# CHAPTER 2 


## [Configuración del entorno para el desarrollo Jupyter de Anaconda](https://read.amazon.com/?asin=B0BBB56L4X&ref_=kwl_kr_iv_rec_1)


## Secuencia de enseñanza del cuaderno "2_chapter.ipynb" — Capítulo 2: Configuración del entorno de desarrollo



El cuaderno pertenece al curso **Fundamentos de Programación** (código 314029) del profesor Marco Cañas, y sigue esta progresión pedagógica:

**1. Apertura y contexto del curso**
- Enlaces a Colab/Kaggle y video de apoyo para la lectura interactiva.
- Vínculo al programa del curso.
- Declaración de los **principios docentes**: modelo de aula invertida, metodología ABPP (Aprendizaje Basado en Problemas y Proyectos), desarrollo del pensamiento computacional (Python, R, Markdown), ciencia abierta/datos públicos, y enfoque TRRS de Duval.

**2. Presentación del capítulo (CHAPTER 2)**
- Título: *Environment Setup for Development* (Configuración del entorno para el desarrollo), con referencia al libro guía.
- **Structure**: adelanta los temas — configuración del entorno Python, instalación de Anaconda, configuración de Jupyter Notebook, y prueba del entorno.
- **Objectives**: al terminar el capítulo el estudiante podrá configurar el entorno local, trabajar con Jupyter Notebook y ejecutar código Python para verificar la instalación.

**3. Desarrollo práctico paso a paso**
- **Descarga e instalación de Anaconda**: 11 pasos numerados (descarga del instalador, ejecución, aceptación de licencia, ruta de instalación, opciones, finalización).
- **Prueba de la instalación**: verificación de que Anaconda quedó correctamente instalado.
- **Prueba de Python en la terminal Anaconda Prompt**: 4 pasos (abrir la consola, invocar Python, ejecutar comandos simples, salir).
- **Ejecución y prueba de Jupyter Notebook**: pasos para abrir Jupyter desde Anaconda, crear un notebook nuevo y ejecutar una celda de prueba.

**4. Cierre del capítulo**
- **Conclusión** que resume lo aprendido.
- **Preguntas de repaso** (ej. cómo comprobar la versión de Python, cómo abrir la consola interactiva, qué es Jupyter y cómo iniciarlo desde Anaconda).
- Enlace a la **evaluación docente**.
- Enlace de **continuidad** al siguiente capítulo (Capítulo 3: tipos de datos y estructuras básicas).
- Sección de **Referentes** bibliográficos.

En resumen, la secuencia sigue el patrón: *contexto/principios → objetivos y estructura del capítulo → instalación guiada paso a paso (Anaconda → terminal → Jupyter) → verificación práctica → cierre con conclusión, autoevaluación y enlace al siguiente tema*, coherente con el modelo de aula invertida declarado al inicio.

Este capítulo mostrará paso a paso cómo instalar el gestor de paquetes Anaconda y Jupyter Notebook para el desarrollo en Python en un equipo Windows para un proyecto de ciencia de datos. 

Necesitamos el software Python para la instalación; además, necesitamos instalar muchas otras bibliotecas específicas para la tarea. 

Para el análisis y la ciencia de datos, el proyecto [Anaconda](https://www.anaconda.com/download) es muy popular, ya que es fácil de instalar y usar. 

Anaconda es un gestor de paquetes robusto que incluye numerosos paquetes esenciales de código abierto preinstalados (Pandas, NumPy, Matplotlib, etc.). 

In [2]:
!python --version

Python 3.11.9


A lo largo de este libro, utilizaremos la versión 3.11.9 de Python y Jupyter Notebook.

# Structure 

En este capítulo, discutiremos los siguientes temas: 

* Configuración del entorno para el desarrollo de Python 
* Instalación de [Anaconda](https://www.anaconda.com/download) 
* Configuración de Jupyter IPython Notebook 
* Prueba del entorno o la prueba de que lo que instalamos funciona. 

# Objectives   

Después de estudiar este capítulo, podrá:
* Configurar el entorno de desarrollo de Python en el equipo local
* Trabajar con Jupyter Notebook
* Ejecutar código Python para probar la instalación

# Fundamentos de Programacion

## Capítulo 2

Estamos viendo *cómo se trabaja* con **cuaderno jupyter**. 

In [3]:
x = 17 
x 

17


# Downloading and installing the Anaconda package   

Aquí, tenemos los pasos de instalación de Anaconda en la máquina con Microsoft Windows 10. 

## Paso 1: 

Vaya a https://www.anaconda.com/distribution/#download-section, obtendrá la pantalla que se muestra a continuación y haga clic en el botón Descargar(Download Now).  



<img src = 'https://github.com/marco-canas/fundamentos_de_programacion/blob/main/images/anaconda_1.png?raw=true'>

## Paso 2:   

Una vez que haga clic en la página de descarga, comenzará a descargar el archivo .exe de instalación (Anaconda3-2025.12-1-Windows-x86_64).

<img src = 'https://github.com/marco-canas/fundamentos_de_programacion/blob/main/images/anaconda_descarga_2.png?raw=true'> 

En la captura de pantalla anterior, puede ver el inicio de la descarga del archivo .exe de Anaconda. 

## Paso 3:  

Una vez completada la descarga, haga clic derecho en el archivo de instalación (Anaconda3-2025.12-1-Windows-x86_64) y seleccione "Ejecutar como administrador".

<img src = 'https://github.com/marco-canas/fundamentos_de_programacion/blob/main/images/anaconta_ejecutar_como_administrador_3.png?raw=true'>

## Paso 4: 

Haga clic en el botón Siguiente, como se muestra en la siguiente captura de pantalla:  

<img src = ''>

<img src = 'https://github.com/marco-canas/fundamentos_de_programacion/blob/main/images/inicio_instalacion_anaconda_1.jpg?raw=true'>

## Paso 5: 

Haga clic en el botón Acepto después de leer el Acuerdo de licencia.  

<img src = ''>

<img src = ''>

## Paso 6: 

Haga clic en el botón Siguiente después de seleccionar la opción Solo yo/Todos los usuarios, como se muestra a continuación. En este caso, es Todos los usuarios.

<img src = 'https://github.com/marco-canas/fundamentos_de_programacion/blob/main/images/seleccion_tipo_instalacion_2.jpeg?raw=true'>

## Paso 7: 

Ahora, especifique la ruta de la carpeta de instalación y haga clic en el botón Siguiente.

<img src = 'https://github.com/marco-canas/fundamentos_de_programacion/blob/main/images/elegir_localizacion_instalacion_3.jpeg?raw=true'>

## Paso 8: 

Ahora, marque ambas casillas de verificación y haga clic en el botón Instalar.

<img src = 'https://github.com/marco-canas/fundamentos_de_programacion/blob/main/images/opciones_instalaci%C3%B3n_avanzada_4.jpeg?raw=true'>

## Paso 9: 

Tras hacer clic en el botón Instalar, comenzará la instalación. Verá las siguientes pantallas; espere a que finalice la instalación:

<img src = 'https://github.com/marco-canas/fundamentos_de_programacion/blob/main/images/completando_anaconda_prompt_8.jpeg?raw=true'>

## Paso 10: 

Una vez completado, haga clic en el botón Siguiente.

<img src = 'https://github.com/marco-canas/fundamentos_de_programacion/blob/main/images/instalacion_completa_6.jpeg?raw=true'>

## Paso 11: 

Haga clic en el botón Finalizar en la nueva pantalla. Anaconda se instaló correctamente.

Al hacer clic en el botón Finalizar, se abrirá una página web en el navegador con más información sobre el producto Anaconda, que puede ignorar. 

<img src = 'https://github.com/marco-canas/fundamentos_de_programacion/blob/main/images/anaconda_navegator_9.jpeg?raw=true'>

En este punto, hemos completado la instalación de Anaconda. 

Ahora es momento de probarla y comprender el entorno de desarrollo de Python y Anaconda.

# Probando la instalación.   



Tras completar la instalación de Anaconda, comprobaremos la configuración de Python y Jupyter Notebook. 

Para verificar la instalación, siga estos pasos:

# Prueba de Python en la terminal Anaconda Promt  



## Paso 1: 

Ve a Inicio (Parte inferior izquierda de tu computador) y escribe Anaconda Prompt para abrir para abrir la Terminal asociada a Anaconda.  

<img src = 'https://github.com/marco-canas/fundamentos_de_programacion/blob/main/images/terminal_anaconda_prompt.png?raw=true'>

## Paso 2: 

Para comprobar si Python está instalado, escriba: `Python –-version` en el símbolo del sistema y pulse la tecla Enter.   
Si obtiene un resultado como el de la siguiente captura de pantalla, significa que Python se instaló correctamente:



<img src = 'python_version_desde_anaconda_prompt.png'>

## Paso 3: 

Ahora, escribe "python" y pulsa Enter para iniciar el Interactive Python Shell. 

Obtendrás un resultado como el de la siguiente captura de pantalla:  



<img src = 'verificar_python_en_anaconda_prompt.png'>

## Paso 4: 

Ahora escriba `print('Fundamentos de programación')` y presione Intro para ejecutar la instrucción print. 

Si la instalación se realizó correctamente, obtendrá un resultado como el de la siguiente captura de pantalla:

<img src = 'imprimir_fundamentos_de_programacion.png'> 

## Paso 4: 

Para salir del shell interactivo de Python, escriba `quit()` (abandonar) y presione Enter.



<img src = 'https://github.com/marco-canas/fundamentos_de_programacion/blob/main/images/quit_8.png?raw=true'> 

Ya hemos visto cómo ejecutar el código Python con el intérprete interactivo de Python. 

Veamos cómo podemos usar Jupiter Notebook para ejecutarlo.

# Ejecución y prueba de Jupyter Notebook  

Jupyter Notebook es una plataforma popular para escribir y ejecutar código Python entre científicos y analistas de datos.

Esta sección del capítulo mostrará cómo ejecutar Jupyter Notebook y el código Python. 

## Paso 1: 



Primero, En la terminal Anaconda Prompt creemos un directorio de trabajo (una carpeta simple de Windows) escribiendo el siguiente comando en Anaconda Prompt: `mkdir fundamentos_de_programacion_con_python`   


<img src = 'crar_una_carpeta_con_mkdir.png'>

## Paso 2: 

Luego, cambia el directorio escribiendo `cd fundamentos_de_programacion_con_python`. 

## [Paso 3:](https://read.amazon.com/?asin=B0BBB56L4X&ref_=kwl_kr_iv_rec_1) 



Ahora, escriba `jupyter notebook` en este Anaconda Prompt y presione Enter.  



<img src = 'https://github.com/marco-canas/fundamentos_de_programacion/blob/main/images/jupyter_notebook_10.png?raw=true'>

Iniciará el servidor local y obtendrá una página web de Jupyter Notebook como se muestra a continuación.  

<img src = 'pagina_web_de_jupyter_notebook.png'>

## Paso 4: 

Cambia el clic en el nuevo botón desplegable en la parte superior derecha y selecciona Python3.

<img src = 'seleccion_kernel_python_3.png'>

## Paso 5: 

Verá una página como la de la siguiente captura de pantalla, donde cada fila se denomina celda.   

Podemos agregar y eliminar celdas usando la opción del menú Archivo.

<img src = ''>

## Paso 4: 

Ahora, escribiremos y ejecutaremos la instrucción de impresión de Python.   

Primero, escriba el código Python que se muestra a continuación en la celda y, para ejecutarlo, presione Mayús+Intro (o use la opción de menú).  

Se ejecutará el código y obtendrá lo siguiente: `print("Bienvenido al curso de Fundamentos de programación")`

<img src = 'print_en_celda_cuaderno.png'>

Si ha completado con éxito todos los pasos mencionados anteriormente, significa que ha instalado correctamente el paquete Anaconda para el desarrollo en Python.

## Conclusión:  



En este capítulo, instalamos y probamos el entorno de desarrollo Anaconda-Python. 

Existen muchos IDE disponibles para el desarrollo en Python en el mercado. 

La elección del IDE depende totalmente del desarrollador; depende de su conveniencia y preferencia. 

En general, la mayoría de los científicos y analistas de datos utilizan Jupyter Notebook para su desarrollo inicial. 

En el siguiente capítulo, aprenderemos los fundamentos de la programación en Python con ejemplos prácticos de codificación. 

## Preguntas:   

* ¿Qué es Anaconda?   
* Lista de algunos paquetes/bibliotecas preinstalados en Anaconda. 
* ¿Cómo comprobar la versión de Python instalada? 
* ¿Cómo abrir la consola interactiva de Python? 
* ¿Qué es Jupyter Notebook y cómo se puede iniciar mediante Anaconda Prompt?

### [Evaluamos al profesor Marco Cañas Aquí](https://forms.office.com/Pages/ResponsePage.aspx?id=IefhmYRxjkmK_7KtTlPBwkanXIs1i1FEujpsZgO6dXpUREJPV1kxUk1JV1ozTFJIQVNIQjY5WEY3US4u)

### Continue su aprendizaje en la siguiente clase a través del siguiente [vínculo](https://github.com/marco-canas/fundamentos_de_programacion/blob/main/clases/unidad1/3_chapter.ipynb)

[CAPÍTULO 3. Sección 3.1 Tipos de datos integrados y Estructuras de datos básicas](https://github.com/marco-canas/fundamentos_de_programacion/blob/main/2_clases/unidad1/3_chapter_section_1_tipos_datos_estructuras_incorporadas.ipynb)

## Referentes 

[Libro guía Fundamentos de programación: Data Analysis with Python: Introducing NumPy, Pandas, Matplotlib, and Essential Elements of Python Programming (English Edition) [Kindle Edition]](https://read.amazon.com/?asin=B0BBB56L4X&ref_=kwl_kr_iv_rec_1)